apps/ui/src/components/Modal/ChangeController.vue

```diff
@@ -0,0 +1,115 @@
+<script setup lang="ts">
+import { clone } from '@/helpers/utils';
+import { getValidator } from '@/helpers/validation';
+
+const DEFAULT_FORM_STATE = {
+  controller: ''
+};
+
+const props = defineProps<{
+  open: boolean;
+  initialState?: any;
+}>();
+
+const emit = defineEmits<{
+  (e: 'save', value: string);
+  (e: 'close');
+}>();
+
+const CONTROLLER_DEFINITION = {
+  type: 'string',
+  format: 'ens-or-address',
+  title: 'Controller',
+  examples: ['Address or ENS']
+};
+
+const formValidator = getValidator({
+  $async: true,
+  type: 'object',
+  additionalProperties: false,
+  required: ['controller'],
+  properties: {
+    controller: CONTROLLER_DEFINITION
+  }
+});
+
+const form: {
+  controller: string;
+} = reactive(clone(DEFAULT_FORM_STATE));
+const formValidated = ref(false);
+const showPicker = ref(false);
+const searchValue = ref('');
+const formErrors = ref({} as Record<string, any>);
+
+watch(
+  () => props.open,
+  () => {
+    if (props.initialState) {
+      form.controller = props.initialState.controller;
+    } else {
+      form.controller = '';
+    }
+  }
+);
+
+watchEffect(async () => {
+  formValidated.value = false;
+
+  formErrors.value = await formValidator.validateAsync(form);
+  formValidated.value = true;
+});
+</script>
+
+<template>
+  <UiModal :open="open" @close="$emit('close')">
+    <template #header>
+      <h3>Change controller</h3>
+      <template v-if="showPicker">
+        <button
+          type="button"
+          class="absolute left-0 -top-1 p-4"
+          @click="showPicker = false"
+        >
+          <IH-arrow-narrow-left class="mr-2" />
+        </button>
+        <div class="flex items-center border-t px-2 py-3 mt-3 -mb-3">
+          <IH-search class="mx-2" />
+          <input
+            ref="searchInput"
+            v-model="searchValue"
+            type="text"
+            placeholder="Search name or paste address"
+            class="flex-auto bg-transparent text-skin-link"
+          />
+        </div>
+      </template>
+    </template>
+    <template v-if="showPicker">
+      <PickerContact
+        :loading="false"
+        :search-value="searchValue"
+        @pick="
+          form.controller = $event;
+          showPicker = false;
+        "
+      />
+    </template>
+    <div v-else class="s-box p-4">
+      <UiInputAddress
+        v-model="form.controller"
+        :definition="CONTROLLER_DEFINITION"
+        :error="formErrors.delegatee"
+        @pick="showPicker = true"
+      />
+    </div>
+    <template v-if="!showPicker" #footer>
+      <UiButton
+        class="w-full"
+        :disabled="Object.keys(formErrors).length > 0"
+        @click="emit('save', form.controller)"
```

Comment:
sometimes we are using `$emit` and sometimes we are using `emit` 🤔 i am not too sure what's the difference

Reply:
I think we should always use `$emit` inside `<template>`, and `emit` inside `<script>`.

I believe we got a similar debate about that with Sam some time ago

Reply:
The only difference is that `$emit` is special keyword that you can use in Vue templates out of the box, while `emit` can be only used if you assign it in script (it could actually be called anything).
```ts
const emit = defineEmits(...)
// you can name it whatever you want and use it in templates via that name
```

I think I usually end up using named emit vs special keyword, but in most cases I end up using the same format that the file I copied source from 😆 

I don't see any eslint rules to enforce it.

Reply:
I remember that we've already got confused once https://github.com/snapshot-labs/sx-monorepo/pull/100#discussion_r1508844944

Maybe just always assign the emit to a constant, and avoid the magical `$emit` from now ?

Reply:
> Maybe just always assign the emit to a constant, and avoid the magical $emit from now ?

That's what I'd prefer, but we can discuss it and later on convert the codebase. Might be annoying to enforce without eslint rule for it.